### DISCUSSION TITLE
Feature Selection Issue

### DISCUSSION THREAD

**Discussion by Snowcatcat** [facebookexperimental/Robyn#293] [Q&A]
Hi,

I'm wondering if there is any feature selection algorithm built into Robyn, if not yet, what approach would you recommend to select the most relevant features? Should I do stepwise/forward/backward selection somewhere else first then input the selected variables into Robynn to do a final checkout? 

What about when we also need to transform the external variables (context_vars), such as adding polynomial terms?

**Comment by gufengzhou** [comment]
Hi sorry for the delay. Robyn's built-in selection mechanism is ridge/L2 regularisation, which in practise often shrinks beta coefficients for variables that lack variation and thus "selects". But of course you can prepare your input data with any desired selection techniques. Same for transformation that is also rather subjected to individual needs.
